components/query-results.js

```diff
@@ -50,23 +63,34 @@ const queryResults = {
               this.resText =
                 "<pre>" + JSON.stringify(res.data, undefined, 2) + "</pre>";
               if (data.length) {
-                const colnames = Object.keys(data[0]);
-                this.tableData = data;
-                this.columns = colnames.map((c) => {
-                  return {
-                    prop: c,
-                    label: c,
-                    minWidth: "20px",
-                  };
-                });
-              } else {
-                for (let x in data) {
-                  outputHTML += x + " | " + data[x] + "<br />";
+                if (data[0] instanceof Object) {
+                  const colnames = Object.keys(data[0]);
+                  this.tableData = formatData(colnames, data);
+                  this.columns = colnames.map((c) => {
```

Comment:
Quick tip, you can simplify further:

~~~
map((c) => {
 return { foo: "bar" };
});
~~~

is equivalent to:

~~
map((c) => ({ foo: "bar" }));
~~~

Reply:
Thanks!